### ISSUE TITLE
Game loops instead of showing a hint text

### ISSUE THREAD

**Issue by un-logic** [00-Evan/shattered-pixel-dungeon-gdx#21]
Steps to reproduce:
1a. Start a game and immediately press "Enter" or on tile with hero.
2a. Start a game and immediately press on tile with sign.
3a. During a game, place your character outside of the room with ladder and sign;
3b. Press on tile with a sign.
This sometimes works with movement keys also.
_Game version_ 0.4.2b.
_JRE build_ 1.8.0_102 on Windows 7 Pro SP1.

**Comment by 00-Evan** [comment]
This seems to be caused by the threading improvements in 0.4.2.

It's affecting both the mobile and desktop versions, but the desktop version is giving a more pronounced symptom.

Will fix, thanks.

**Comment by 00-Evan** [comment]
This seems to affect multiple elements of the game, not just menus, previously was going to include this fix in 0.4.3 but it seems significant enough to warrant a separate desktop-specific patch.

**Comment by 00-Evan** [comment]
fixed in commit 12647bd846d7f4a9a879683309e5fbe3686b6647

You should now be able to download v0.4.2c, which is desktop exclusive, and fixes this issue.

**Comment by un-logic** [comment]
Decided not to file a new issue, 'cause my gut feeling is this new glitch is related to this one.
In recently updated 0.4.2c after my hero stepped inside a room behind a locked door screen turned all white and some rare flickering started all over it. After few attempts of screen capturing involving hitting Alt+Tab rare flickering turned gradually into a dense scattering of UI elements. Take a look:
![panorama](https://cloud.githubusercontent.com/assets/9430664/19103284/f7644852-8ae6-11e6-86e6-e39fea1c8cb3.png)
Then, after I moved a hero one step in random direction it became to look like this:
![aftermath](https://cloud.githubusercontent.com/assets/9430664/19103510/47d492be-8ae8-11e6-867d-ad6b652023ac.png)
Reloading the game via main menu fixed this issue. Can't reproduce so far.

**Comment by 00-Evan** [comment]
That's very odd. It may be related to 0.4.2 changes but shouldn't be tied to the specific issue present with the window background.

Let me know if you're able to reproduce and I'll look into it.

**Comment by un-logic** [comment]
@00-Evan, it did reoccurred, please, notice the update on my previous message.
Also feel free to detach this issue to a standalone.

**Comment by 00-Evan** [comment]
The two issues you're describing are not necessarily related. One is strictly tied to the game's camera while the other appears to be texturing and draw coordinate related.

Regardless both of these seem really random and there's nothing I can do about them without reproduction steps.

**Comment by un-logic** [comment]
@00-Evan, I recently experienced this issue again.
Now I can definitely say that these two issues are one.
So, my character picked up a potion in a room near by alchemy pot, and everything started flickering, just like the first time. This time I've waited for a while and noticed that flickering become to reveal some pattern. You would call it **texturing and draw coordinate** problem. But after some 30 seconds puzzle of textures "combined" and it started to look like if camera was shaking hard. What you called **game's camera** problem.
So to me it's clearly related to camera positioning, which was quite obvious from shaking issue, while "textures scattering" is just an extreme form of it, that transits to shaking and then dies out.
Unfortunately, there seems like no system in occurring of this bug. Purely random occasions.
![aftermath2](https://cloud.githubusercontent.com/assets/9430664/19231635/2a05426c-8eec-11e6-86fd-8d79e87a1a9e.png)
